pipeline.go

```diff
@@ -196,51 +196,51 @@ func (p *Pipeline) Run() (PipelineResult, error) {
 //
 // If a specific error handler has not been defined, the default behavior is to
 // return the PipelineResult until the last stage executed and the error occurred.
-func handleError(result *PipelineResult, ser StageExecutionResult, previousStatus status.Code, msg string, handler Stage) (PipelineResult, error) {
-	ser.FinalTime = time.Now()
-	result.StagesResults = append(result.StagesResults, ser)
+func handleError(result *PipelineResult, previousSer StageExecutionResult, previousStatus status.Code, msg string, handler Stage) (PipelineResult, error) {
+	previousSer.FinalTime = time.Now()
+	result.StagesResults = append(result.StagesResults, previousSer)
 
 	if handler.Dir != "" {
 		var serError StageExecutionResult
 		var err error
 		serError.Stage = handler.Name
 		serError.StartTime = time.Now()
 
-		id := fmt.Sprintf("%s/%s calls Error Handler", result.Name, ser.Stage)
+		id := fmt.Sprintf("%s/%s calls Error Handler", result.Name, previousSer.Stage)
 		serError.BuildResult, err = buildImage(id, handler.Dir, handler.BuildEnv)
 		if err != nil {
 			result.StagesResults = append(result.StagesResults, serError)
-			result.Status = status.ErrorHandlerError
+			result.Status = status.Text(status.ErrorHandlerError)
 			result.FinalTime = time.Now()
 
-			return *result, fmt.Errorf("error when building image for error handler: %s", err)
+			return *result, status.NewError(status.BuildError, fmt.Errorf("error when building image for error handler: %s", err))
 		}
 		if status.Code(serError.BuildResult.ExitStatus) != status.OK {
 			result.StagesResults = append(result.StagesResults, serError)
-			result.Status = status.ErrorHandlerError
+			result.Status = status.Text(status.ErrorHandlerError)
 			result.FinalTime = time.Now()
 
-			return *result, fmt.Errorf("error when building image for error handler: Status code %d(%s) when running image for %s", serError.BuildResult.ExitStatus, status.Text(status.Code(serError.BuildResult.ExitStatus)), id)
+			return *result, status.NewError(status.BuildError, fmt.Errorf("error when building image for error handler: Status code %d(%s) when running image for %s", serError.BuildResult.ExitStatus, status.Text(status.Code(serError.BuildResult.ExitStatus)), id))
 		}
 
-		serError.RunResult, err = runImage(id, handler.Dir, string(previousStatus), handler.RunEnv)
+		serError.RunResult, err = runImage(id, handler.Dir, status.Text(previousStatus), handler.RunEnv)
```

Comment:
Ao revisar esse pedaço de código fiquei em dúvida sobre qual seria o stdout que deveríamos passar para runImage do HandlerError. No fluxo padrão sempre passamos o `result.StagesResults[index-1].RunResult.Stdout `.

Nesse caso deveríamos passar o Stderr ou seguir o que fazemos no fluxo padrão? 
Me parece que o segundo faz mais sentido considerando o objetivo de persistir os arquivos extraídos em um estágio anterior. Nesse caso, teríamos que considerar o ultimo estágio válido mesmo, pois se considerarmos o atual ele pode não ter informações a serem persistidas.

Reply:
O estágio handler error tem que receber via STDIN uma versão serializada JSON de todas as informações de execução do estágio que deu erro (`StageExecutionResult`). É um STDIN fixo, independente de quem for implementar ele será sempre o mesmo.